source/Octopus.Client/IOctopusAsyncClient.cs

```diff
@@ -354,9 +354,9 @@ public interface IOctopusAsyncClient : IDisposable
         /// <summary>
         /// Get a repository for the given space
         /// </summary>
-        /// <param name="spaceId"></param>
+        /// <param name="space"></param>
         /// <returns></returns>
-        IOctopusSpaceAsyncRepository ForSpace(string spaceId);
+        IOctopusSpaceAsyncRepository ForSpace(SpaceResource space);
```

Comment:
The more I think about it, the more I realise this is probably the right way to go :+1:. I presume this change will also require a bunch of test setup changes in octopus server. Probably best to do those changes before you merge this PR to avoid blocking anyone else making changes in this repo.

Reply:
I didn't modify `RepositoryScope.ForSpace(string spaceId)` should I?

Reply:
done